### DISCUSSION TITLE
External module from an entry point using library

### DISCUSSION THREAD

**Discussion by RonenISRb** [webpack/webpack#14954] [Q&A]
Hello :)
I'd like to got help regarding the following question.
I have a module which is a configuration that's imported in different modules in my project (chrome extension).
I want it to be 'hardcoded' meaning that after the bundling, we can edit it manually and pack as CRX, and the actual config will be read on runtime, once the extensions will be enabled in the browser.

Because of this requirement, I don't wont this module to be 'bundled' with each file that's importing it, since I need it to be as a separate file which can  modified easilly, and should be used by other modules.

How can I achieve this goal?

I tried the following steps but it didnt work:
1) Added new entry point in my webpack.config and defined the config module with library name and type:'commonjs-module'.
```
entry: {
        ...
        config: { 
            import: path.join(dir'hardcoded_configuration/config.ts'),
            library: {
                name: 'ExtensionConfig',
                type: 'commonjs-module'
            }
        }
    },
```

2) Modified the 'resolve' config to include the path of the config module which is created using step 1
```
    resolve: {
        modules: [path.resolve('./node_modules'), path.resolve('./hardcoded_configuration')],
        alias: alias,
        extensions: [".ts", ".js", ".tsx", ".jsx"],
    },
```

3) Modify the 'externals':
```
    externals: {
        extensionConfig: 'ExtensionConfig',
    },
```

4) Importing it in the modules:
`import * as config from 'extensionConfig'
`
But it doesn't work :(

Can you help please on this issue? What am I missing? Can it be achieved?

**Comment by RonenISRb** [comment]
Someone :)?

**Reply by alexander-akait** [reply]
Can you create simple reproducible example?

**Reply by RonenISRb** [reply]
@alexander-akait 
I can try do it, but the question is also if it's possible to accomplish my need, and if yes - whether the way I described is correct?

Not bundling several existing modules (with the modules which reference them), but:
1) for every such module that I want to no be bundled inside the referenced files, configure it as an entry point in the webpack with 
```
 hardcoded_config: { 
            import: path.join(dir,'hardcoded_configuration/config.ts'),
            library: {
                name: 'ExtensionConfig',
                type: 'commonjs-module'
            }
        }
```
2)declare them as 'externals' and change the import methodology in the modules which use them.

**Reply by alexander-akait** [reply]
Sounds like you need `externals`, I do not understand very well what you mean by `edit it manually and pack as CRX`, so it is hard to answer
